### DISCUSSION TITLE
Received header vs Date header in ThirdStats reports

### DISCUSSION THREAD

**Discussion by Chronophan** [devmount/third-stats#360] [Q&A]
The default Thunderbird folder view includes a Date column based on the email's Date header field. Unfortunately, the Date header may contain an incorrect timestamp for various reasons, from system clock malfunction or wrong time zone setting on the originating system to deliberate forging by spammers. The Received column, if enabled, also displays the Date header timestamps, effectively rendering itself useless.   It appears that ThirdStats also pulls the Received date from the Date header.

I'm using ThirdStats to track Spam trends, so it's picking up a big spike at midnight since spammers often forge just a date in the Date fields of their spam.  If ThirdStats uses the Date header field data, please let me know if there is an easy way to change the source to the Received header field.

Background:
Thunderbird has an advanced setting that corrects this problem by populating the Received column with the Received header field content that I learned about here:
http://blog.dmitryleskov.com/small-hacks/putting-the-received-column-in-thunderbird-to-work/
Although I applied this fix and updated the folders so that the Received column now matches the Received header fields, it appears that ThirdStats is still pulling from the forged Date headers.

**Comment by devmount** [comment]
Hi @Chronophan, many thanks for reporting this, I didn't know about this date field issue. ThirdStats makes use of Thunderbirds Webextension APIs to retrieve the message data, including the message date. As you can see [in the docs](https://webextension-api.thunderbird.net/en/latest/messages.html#messageheader), there is currently only one single `date` field and no possibility to control which date to use.

However, I created a corresponding thread in Thunderbirds topicbox:
https://thunderbird.topicbox.com/groups/addons/T3db2fad2056c3809/received-header-vs-date-header

Let's hope that somebody has some insights about this problem and maybe the API will get an update to take this advanced setting you mentioned into account.
